### PR TITLE
feature: updates pydantic usage to work with 2.1

### PR DESCRIPTION
**Pull request by jaydrennan**

### CODE REVIEW

File: imaginairy/schema.py [hunk 1]
```diff
@@ -159,26 +182,26 @@ def __repr__(self):
 
 
 class ImaginePrompt(BaseModel):
-    prompt: Optional[List[WeightedPrompt]]
-    negative_prompt: Optional[List[WeightedPrompt]]
+    prompt: Optional[List[WeightedPrompt]] = None
+    negative_prompt: Optional[List[WeightedPrompt]] = None
```
Comment (jaydrennan) [inline imaginairy/schema.py:186]: I tried to use the auto upgrade tool they had. Didn't do much. It added all these " = None " here.

Comment (jaydrennan) [reply imaginairy/schema.py:186]: I didn't try running without them

File: imaginairy/schema.py [hunk 2]
```diff
@@ -199,7 +222,8 @@ def __init__(self, prompt=None, **kwargs):
         # allows `prompt` to be positional
         super().__init__(prompt=prompt, **kwargs)
 
-    @validator("prompt", "negative_prompt", pre=True, always=True)
```
Comment (jaydrennan) [inline imaginairy/schema.py:202]: I didn't really understand their explanation of the removal of "always" and what should be done to replicate behavior.
Screenshot 2023-08-08 at 8 43 44 PM

Comment (jaydrennan) [reply imaginairy/schema.py:202]: Seems contradictory?:

Screenshot 2023-08-09 at 3 14 03 AM

**Comment by brycedrennan** [comment]
@jaydrennan just started looking at this. is it in a working state?

**Comment by jaydrennan** [comment]
@brycedrennan Yes, I can generate an image successfully with the changes I've made here. Haven't run other features

**Comment by brycedrennan** [comment]
Thanks for taking a stab at it. It was more a lot more involved to get it all working so I started from scratch so I could know what was causing test failures with incremental changes.  Finally got it working here: https://github.com/brycedrennan/imaginAIry/pull/378

**Pull request closed without merging**